lang/test/org/partiql/lang/TestBase.kt

```diff
@@ -16,213 +16,15 @@
-
-    this.assertThat(ex.errorCode).isEqualTo(errorCode)
-
-    val errorContext = ex.errorContext
-
-    correctContextKeys(errorCode, errorContext)
-    correctContextValues(errorCode, errorContext, expectedValues)
-}
-
-/**
- * Checks that [errorContext] contains keys for each [Property] found in [errorCode]
- *
- * @param errorCode for the exception thrown by the test
- * @param errorContext errorContext that was part of the exception thrown by the test
- */
-private fun SoftAssertions.correctContextKeys(errorCode: ErrorCode, errorContext: PropertyValueMap?): Unit =
-    errorCode.getProperties().forEach {
-        assertThat(errorContext!!.hasProperty(it))
-            .withFailMessage("Error Context does not contain $it")
-            .isTrue
-    }
-
-/**
- * Check that for the given [errorCode], [errorContext] thrown by the test, it contains all the [expected] values
- * specified by the test.
- *
- * @param errorCode for the exception thrown by the test
- * @param errorContext errorContext that was part of the exception thrown by the test
- * @param expected expected values for errorContext
- */
-private fun SoftAssertions.correctContextValues(errorCode: ErrorCode, errorContext: PropertyValueMap?, expected: Map<Property, Any>) {
-
-    assertThat(errorCode.getProperties().containsAll(expected.keys))
-        .withFailMessage(
-            "Actual errorCode must contain these properties: " +
-                "${expected.keys.joinToString(", ")} but contained only: " +
-                errorCode.getProperties().joinToString(", ")
-        )
-        .isTrue
-
-    val unexpectedProperties = errorCode.getProperties().filter { p -> !expected.containsKey(p) }
-    if (unexpectedProperties.any()) {
-        fail("Unexpected properties found in error code: ${unexpectedProperties.joinToString(", ")}")
-    }
-
-    if (errorContext == null) return
-    expected.forEach { entry ->
-        val actualPropertyValue: PropertyValue? = errorContext[entry.key]
-        assertThat(errorContext.hasProperty(entry.key))
-            .withFailMessage("Error Context does not contain ${entry.key}")
-            .isTrue
-
-        val message by lazy {
-            "Expected property ${entry.key} to have value '${entry.value}' " +
-                "but found value '$actualPropertyValue'"
-        }
-
-        val propertyValue: Any? = actualPropertyValue?.run {
-            when (entry.key.propertyType) {
-                PropertyType.LONG_CLASS -> longValue()
-                PropertyType.STRING_CLASS -> stringValue()
-                PropertyType.INTEGER_CLASS -> integerValue()
-                PropertyType.TOKEN_CLASS -> tokenTypeValue()
-                PropertyType.ION_VALUE_CLASS -> ionValue()
-            }
-        }
-
-        assertThat(propertyValue)
-            .withFailMessage(message)
-            .isEqualTo(entry.value)
-    }
-}
 
 @RunWith(JUnitParamsRunner::class)
 abstract class TestBase : Assert() {
```

Comment:
The old `TestBase` class contained a bunch of protected functions that weren't being used except in a small handful of places anymore so I decided to remove them and refactor the uses.

Reply:
It's also worth noting that `TestBase` still inherits from `org.junit.Assert`, which is how many of the tests access the `assert*` methods (through `Assert`'s `protected` methods.)  I really dislike this since these functions shadow anything `import`ed, which creates some confusing situations sometimes in the IDE's code completion.  This also means we have many JUnit5 tests that are using the JUnit4 assertion functions.  This is not a huge problem, just a bit messy.  I considered fixing that to be out of scope for this work, which is why `TestBase` wasn't removed.

Reply:
See #576 and #577.

Reply:
> See #576 and #577.
Would be great to add the refs to the code.